packages/react-components/src/components/datatree-output-component.tsx

```diff
@@ -129,6 +182,54 @@ export class DataTreeOutputComponent extends AbstractOutputComponent<AbstractOut
         }
     }
 
+    private onContextMenu = (event: React.MouseEvent<HTMLDivElement>, id: number): void => {
+        event.preventDefault();
+        event.stopPropagation();
+        this.doContextMenu(event, id);
+    };
+
+    private async doContextMenu(event: React.MouseEvent<HTMLDivElement>, id: number): Promise<void> {
+        if (this.state.xyTree) {
+            const timeProperties: { [key: string]: string } = {};
+            const entry = this.state.xyTree.find(e => e.id === id);
+            if (entry && this.state.columns && this.state.columns.length > 0) {
+                const cols = this.state.columns;
+                for (let i = 0; i < cols.length; i++) {
+                    if (cols[i].dataType === DataType.TIME_RANGE) {
+                        timeProperties[cols[i].title] = entry.labels[i];
+                    }
+                }
+            }
+
+            if (Object.keys(timeProperties).length > 0) {
+                const { show } = useContextMenu({
+                    id: MENU_ID + this.props.outputDescriptor.id,
+                });
+
+                show(event, {
+                    props: {
+                        data: timeProperties,
+                    },
+                    position: this.getMenuPosition(event)
+                });
+            }
+        }
+    }
+    getMenuPosition(event: React.MouseEvent<HTMLDivElement>): { x: number, y: number } {
```

Comment:
The menu is not visible if clicking on the last row if at the bottom of window. Although it appears that this makes the component temporarily taller so the scrollbar can be used to scroll down a little bit more to access the menu. Since there's a workaround, it can be fixed later.

Reply:
Thanks. I propose to fix it in a follow-up commit.